lua/framework/physics/lua_material.cc

```diff
@@ -297,6 +297,16 @@ MatSetProperty(lua_State* L)
 
         prop->Initialize(file_name);
       }
+      else if (operation_index == static_cast<int>(OperationType::OPENMC_MGXSLIB))
+      {
+        if (num_args != 5)
+          LuaPostArgAmountError("MatSetProperty", 5, num_args);
+
+        const auto file_name = LuaArg<std::string>(L, 4);
+        const auto temperature = LuaArg<double>(L, 5);
```

Comment:
At least for the time being, I think it would be best to make the user specify the temperature. If it's wrong in any way, reading the xs data will fail. This is a tad bit fragile, and I'd like to improve on our temperature handling. I wish the OpenMC mgxs format provided a list of the temperatures in a data file. Definitely something we'll want to work on.